scripts/test_release_artifacts.py

```
@@ -0,0 +1,245 @@
+                    self.workflow,
+                )
+
+    def test_every_attestation_dependency_is_fail_closed(self) -> None:
+        proof_gate = self.workflow.index("- name: Verify attestation")
+        attestation_ids = [*ARCHIVE_ATTESTATION_IDS, *INDEX_ATTESTATION_IDS]
+        for attestation_id in attestation_ids:
+            self.assertIn(f"id: {attestation_id}", self.workflow)
+            self.assertIn(f"steps.{attestation_id}.outcome != 'success'", self.workflow)
+            self.assertLess(self.workflow.index(f"id: {attestation_id}"), proof_gate)
+
+        for name, repository in PLATFORM_ARTIFACTS.items():
```

Ruff (0.16.1)

[warning] Loop control variable `repository` not used within loop body

Rename unused `repository` to `_repository`

(B007)